src/dmd/target.d

```diff
@@ -314,6 +323,204 @@ extern (C++) struct Target
                 break;
         }
     }
+
+    /**
+     Sets CPU Operating System, and optionally C/C++ runtime environment from the given triple
+     e.g.
+        x86_64+avx2-apple-darwin20.3.0
+        x86-unknown-linux-musl-clang
+        x64-windows-msvc
+        x64-pc-windows-msvc
+     */
+    void setTriple(const(char)* _triple)
+    {
+        const(char)[] triple = _triple.toDString();
+        const(char)[] next()
+        {
+            size_t i = 0;
+            const tmp = triple;
+            while (triple.length && triple[0] != '-')
+            {
+                triple = triple[1 .. $];
+                ++i;
+            }
+            if (triple.length && triple[0] == '-')
+            {
+                triple = triple[1 .. $];
+            }
+            return tmp[0 .. i];
+        }
+        void unknown(const(char)[] unk, const(char)* what)
+        {
+            unk.toCStringThen!(p => error(Loc.initial,"unknown %s `%s` for `-target`", what, p.ptr));
+        }
+
+        void parseArch(const(char)[] arch)
+        {
+            int len;
+            if (arch.ptr.startsWith("x86_64"))
+            {
+                len = 6;
+                is64bit = true;
+            }
+            else if (arch.ptr.startsWith("x86"))
+            {
+                len = 3;
+                is64bit = false;
+            }
+            else if (arch.ptr.startsWith("x64"))
+            {
+                len = 3;
+                is64bit = true;
+            }
+            else if (arch.ptr.startsWith("x32"))
+            {
+                len = 3;
+                is64bit = true;
+                isLP64 = false;
+            }
+            else
+            {
+                len = 1;
+                unknown(arch, "architecture");
+            }
+            arch = arch[len-1 .. $-1];
+            if (!arch.length)
+                return;
+
+            switch (arch)
+            {
+                case "+sse2": cpu = CPU.sse2; break;
+                case "+avx":  cpu = CPU.avx;  break;
+                case "+avx2": cpu = CPU.avx2; break;
```

Comment:
I think it's better to use an array instead of this. See: https://github.com/dlang/dmd/pull/12508/files#r636396964.

Reply:
see above